Tasks/MavenAuthenticateV0/mavenutils.ts

```diff
@@ -0,0 +1,200 @@
+    }
+
+    return serverElements;
+}
+
+export function getExternalServiceEndpointsServerElements(input: string) {
+    var serviceConnections = getPackagingServiceConnections(input, ["REPOSITORYID"]);
+    var serverElements: any[] = [];
+    if (!serviceConnections || serviceConnections.length === 0)
+    {
+        return serverElements;
+    }
+
+    tl.debug(tl.loc("Info_GeneratingExternalRepositories", serviceConnections.length));
+    for(let serviceConnection of serviceConnections) {
+        switch (serviceConnection.authType) {
+            case (ServiceConnectionAuthType.UsernamePassword):
+                const usernamePasswordAuthInfo = serviceConnection as UsernamePasswordServiceConnection;
+
+                serverElements.push({
+                    id: serviceConnection.additionalData["REPOSITORYID"],
+                    username: usernamePasswordAuthInfo.username,
+                    password: usernamePasswordAuthInfo.password,
+
+                });
+
+                tl.debug(`Detected username/password credentials for '${serviceConnection.packageSource.uri}'`);
+                break;
+            case (ServiceConnectionAuthType.Token):
+                const tokenAuthInfo = serviceConnection as TokenServiceConnection;
+                serverElements.push({
+                    id: serviceConnection.additionalData["REPOSITORYID"],
+                    configuration: {
+                        httpHeaders: {
+                            property: {
+                                name: 'Authorization',
+                                value: 'Basic ' + tokenAuthInfo.token
+                            }
+                        }
+                    }
+                });
+                tl.debug(`Detected token credentials for '${serviceConnection.packageSource.uri}'`);
+                break;
+            case (ServiceConnectionAuthType.PrivateKey):
+                const privateKeyAuthInfo = serviceConnection as PrivateKeyServiceConnection;
+                serverElements.push({
+                    id: serviceConnection.additionalData["REPOSITORYID"],
+                    privateKey: privateKeyAuthInfo.privateKey,
+                    passphrase: privateKeyAuthInfo.passphrase
+                });
+                tl.debug(`Detected token credentials for '${serviceConnection.packageSource.uri}'`);
+                break;
+            default:
+                throw Error(tl.loc('Error_InvalidServiceConnection', serviceConnection.packageSource.uri));
+        }
+    }   
+
+    return serverElements;
+}
+
+export function readXmlFileAsJson(filePath: string): Q.Promise<any> {
+    return readFile(filePath, 'utf-8')
+        .then(convertXmlStringToJson);
+}
+
+export function writeJsonAsSettingsFile(filePath: string, jsonContent: any): Q.Promise<void> {
+    return writeJsonAsXmlFile(filePath, jsonContent.settings, 'settings');
+}
+
+export function mavenSettingsJsonInsertServer (json: any, serverJson:any): any {
```

Comment:
nit: not clear what this function is trying to do. May be rename to addFeedEntryToSettingsJson?

Reply:
Fixed